transports/janus_rabbitmq.c

```diff
@@ -333,12 +323,22 @@ int janus_rabbitmq_init(janus_transport_callbacks *callback, const char *config_
 			/* Something went wrong... */
 			JANUS_LOG(LOG_FATAL, "Got error %d (%s) trying to launch the RabbitMQ outgoing thread...\n", error->code, error->message ? error->message : "??");
 			g_free(rmq_client);
+			janus_config_destroy(config);
 			return -1;
 		}
 		janus_mutex_init(&rmq_client->mutex);
 		/* Done */
 		JANUS_LOG(LOG_INFO, "Setup of RabbitMQ integration completed\n");
 	}
+	g_free(rmqhost);
+	if(to_janus)
+		g_free((char *)to_janus);
+	if(from_janus)
+		g_free((char *)from_janus);
+	if(to_janus_admin)
+		g_free((char *)to_janus_admin);
+	if(from_janus_admin)
+		g_free((char *)from_janus_admin);
```

Comment:
I don't have figures (didn't do any new test), but there should be several advantages. We had a lot of places where we only slept a sequence of tiny amount of time in a loop waiting for something to happen, as we also needed to keep on eye on when something relevant happened (e.g., session destroyed in the meanwhile, Janus hanging up, etc.). This caused the related threads to wake up continuously, and was against the idea of just going to sleep and really waiting for something to happen without needing a constant poll. This approach should help with that, waking threads only when it's really needed and also reducing context switches as a consequence.